lwt/nbd_lwt_unix.ml

```diff
@@ -54,7 +54,20 @@ let io_complete op fd buffer =
   then Lwt.fail End_of_file
   else return ()
 
-let tls_channel_of_fd fd role () =
+let impatient timeout_seconds f buf =
+  let timeout () =
+    Lwt_unix.sleep timeout_seconds >>= fun () ->
+    let msg = Printf.sprintf "Closing connection due to IO timeout: %f seconds" timeout_seconds in
+    Lwt_log.warning msg >>= fun () ->
+    Lwt.fail_with msg
+  in
+  if timeout_seconds > 0.0 then (
```

Comment:
Should this be a `float option` instead of relying on floating point comparison?

Reply:
This is a good idea, it's better to use the type system for such things.

Reply:
It's an optional parameter in the exposed interface of the module, and it's important to check for a positive value anyway in case someone passes in some zero or negative value.

So if this were an option, what should we do if it were `Some -12.34` or `Some 0.0`? (I suppose the choice is between raising an exception or keeping the current behaviour of not using a timeout.)

Reply:
I was actually worried that due to some rounding error somewhere you would get something between 0.0 and `epsilon_float`, and then just timeout way too soon. I'd prefer to raise an exception if someone passes in a timeout that is `< epsilon_float`, and if you already got the `option` in the exposed interface keep that in the internal implementation too as it seems more robust.

Reply:
Does the "thumbs up" mean "Let's leave it as it is" or does it mean "It should still be an option and let's talk about what to do with non-positive values" or what?

Reply:
Ah, I've just seen @edwintorok's comment about `epsilon_float` and rounding errors.
That seems reasonable. I'll do that.

Reply:
My thumbs up means the latter, regardless whether you use an option or not I think it'd be useful to check and raise an exception if you get `Some x` where `x` is not positive.

Reply:
Actually, it probably makes sense to go a bit bigger than `epsilon_float` which is `2.22044604925031308e-16`.
Maybe `1.0e-9`, one nanosecond. (Or maybe this should be yet another thing defined in the `Consts` module...)

Reply:
Yeah, `Consts.min_timeout` sounds good. Lwt seems to use `gettimeofday` a lot internally, which doesn't have a resolution better than a microsecond, but nanosecond is fine too, some day Lwt might be capable of that :)